ikarus/finiteelements/mechanics/loads/volume.hh

```diff
@@ -0,0 +1,146 @@
+// SPDX-FileCopyrightText: 2021-2024 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+/**
+ * @file loads.hh
+ * @brief Definition of the Loads class for application of loads for finite element analysis.
+ * @ingroup mechanics
+ */
+
+#pragma once
+
+#  include <dune/localfefunctions/impl/standardLocalFunction.hh>
+
+#  include <ikarus/finiteelements/febases/powerbasisfe.hh>
+#  include <ikarus/finiteelements/fehelper.hh>
+#  include <ikarus/finiteelements/ferequirements.hh>
+#  include <ikarus/utils/defaultfunctions.hh>
+
+namespace Ikarus {
+
+  /**
+   * @brief Loads class represents different types of loads that can be applied.
+   *
+   * @ingroup mechanics
+   *
+   * @tparam DisplacementBasedElement The type of the displacement-based finite element.
+   */
+  template <typename Basis_, typename FERequirements_ = FERequirements<>, bool useEigenRef = false>
+  class Volume : public PowerBasisFE<typename Basis_::FlatBasis> {
+  public:
+      using Basis                   = Basis_;
+      using FlatBasis               = typename Basis::FlatBasis;
+      using BaseDisp                = PowerBasisFE<FlatBasis>;  // Handles globalIndices function
+      using FERequirementType       = FERequirements_;
+      using LocalView               = typename FlatBasis::LocalView;
+      using Element                 = typename LocalView::Element;
+      using Geometry                = typename Element::Geometry;
+      using Traits                  = TraitsFromLocalView<LocalView, useEigenRef>;
+      using GridView                = typename FlatBasis::GridView;
+      using LocalBasisType          = decltype(std::declval<LocalView>().tree().child(0).finiteElement().localBasis());
+      static constexpr int worldDim = Traits::worlddim;
+
+    /**
+     * @brief Constructor for the Loads class.
+     *
+     * @param p_ele The element on which the load is applied.
+     */
+    template <typename VolumeLoad = utils::LoadDefault>
+    explicit Volume(const Basis& globalBasis, const Element& element, VolumeLoad p_volumeLoad = {}) : BaseDisp(globalBasis.flat(), element) {
+        this->localView().bind(element);
```

Comment:
All this construction here is maybe not needed. Maybe you can get here the displacement based element and store that here. Then you can later call within the code `dispFe.geometry()`or `dispFe.numberOfNodes()` or similar.
Try to reduce code duplciation